### ISSUE TITLE
[secp256r1] Use the p256 crate's implementation of recoverable signatures

### ISSUE THREAD

**Issue by jonas-lj** [MystenLabs/fastcrypto#510]
Since v0.13.0, the p256 crate supports recoverable signatures. There's as of v0.13.0 a todo in their code about handling reduced x's which is not handled yet, but when that's done, we should consider using their implementation instead of our own.

**Comment by jonas-lj** [comment]
This is no longer relevant after #566
